blog_post_collaboration/kedro_ibis.md

```diff
@@ -0,0 +1,263 @@
+
+If you want to peek at the final solution using Kedro at this point, [jump to the "Quickstart" section](#quickstart).
+
+### Creating a custom `ibis.Table` dataset
+
+For our Kedro-Ibis integration, we need the ability to load and save data using the backends Ibis supports. The data will be represented in memory as an [Ibis table](https://ibis-project.org/reference/expression-tables#ibis.expr.types.relations.Table), analagous to a pandas dataframe. The first step in using Ibis involves connecting to the desired backend; for example, see [how to connect to the DuckDB backend](https://ibis-project.org/backends/duckdb#connect). We abstract this step in the dataset configuration. We also maintain a mapping of established connections for reuse—a technique borrowed from existing dataset implementations, like [those of `pandas.SQLTableDataset` and `pandas.SQLQueryDataset`](https://github.com/kedro-org/kedro-plugins/blob/kedro-datasets-2.0.0/kedro-datasets/kedro_datasets/pandas/sql_dataset.py).
+
+We also need to support different [materialisation strategies](https://docs.getdbt.com/docs/build/materializations). Depending on what the user configures, we call either [`create_table`](https://ibis-project.org/backends/duckdb#ibis.backends.duckdb.Backend.create_table) or [`create_view`](https://ibis-project.org/backends/duckdb#ibis.backends.duckdb.Backend.create_view).
+
+[Find the complete dataset implemention on GitHub.](https://github.com/deepyaman/jaffle-shop/blob/main/src/jaffle_shop/datasets/ibis/table_dataset.py)
+
+### Configuring backends with the `OmegaConfigLoader` using variable interpolation
+
+Since Kedro 0.18.10, you can [leverage OmegaConf-native templating in catalog files](https://docs.kedro.org/en/latest/configuration/advanced_configuration.html#catalog). In our example:
+
+```yaml
+# conf/base/catalog_connections.yml
+_duckdb:
+  backend: duckdb
+  # `database` and `threads` are parameters for `ibis.duckdb.connect()`.
+  database: jaffle_shop.duckdb
+  threads: 24
+
+# conf/base/catalog.yml
+raw_customers:
+  type: jaffle_shop.datasets.ibis.TableDataset
+  table_name: raw_customers
+  connection: ${_duckdb}
+  save_args:
+    materialized: table
+
+...
+```
+
+### Building pipelines
+
+Develop pipelines the same way you would any other Kedro pipeline. Below, we show a node written using Ibis alongside the original dbt model. Note that Ibis solves the [parametrisation problem](https://www.youtube.com/watch?v=XdZklxTbCEA&t=514s) with vanilla Python syntax; on the other hand, the dbt counterpart uses Jinja templating.
```

Comment:
```suggestion
Develop pipelines the same way you would any other Kedro pipeline. Below, you can see a node written using Ibis alongside the original dbt model. Note that Ibis solves the [parametrisation problem](https://www.youtube.com/watch?v=XdZklxTbCEA&t=514s) with vanilla Python syntax; on the other hand, the dbt counterpart uses Jinja templating.
```

better to avoid first person when possible? cc @stichbury (I always guide myself by https://ddbeck.com/first-person-plural-is-questionable/)

Reply:
Will wait for Jo's input on this, since I probably need to change it more than just this place then.

Reply:
I don't have a particular view on first person plural, and personally we think "we" is fine if used sparingly :)

But it's a bit weird in that some of this piece uses "we" and some uses "you", viz these two sentences right here: 
> Develop pipelines the same way you would any other Kedro pipeline. Below, we show a node written using Ibis alongside the original dbt model.

I would tend to remove some of the "you" in this case, so rewrite as follows:

> Develop pipelines the same way you would any other Kedro pipeline. The code below shows a node written using Ibis alongside the original dbt model.

You don't need to contort all the text to remove "you" but perhaps have a scan through the middle method section and attempt to eliminate that, keeping the you in the top and tail, which address the reader more directly, and retaining just "we" for the text that shows the steps? I can have a go if you like, but I'll let you have first dibs.

Reply:
> I can have a go if you like, but I'll let you have first dibs.

You're welcome to have a go! If you haven't gotten to it yet, I'll try later today or tomorrow, but I trust whatever you do will be at least as good (and almost certainly better). :)